pkgs/development/libraries/science/math/mkl/default.nix

```diff
@@ -0,0 +1,55 @@
+{ stdenvNoCC, writeText, fetchurl, rpm, cpio, openmp }:
+
+stdenvNoCC.mkDerivation rec {
+  name = "mkl-${version}";
+  version = "${date}.${rel}";
+  date = "2019.0";
+  rel = "117";
+
+  src = fetchurl {
+    url = "http://registrationcenter-download.intel.com/akdlm/irc_nas/tec/13575/l_mkl_${version}.tgz";
+    sha256 = "1bf7i54iqlf7x7fn8kqwmi06g30sxr6nq3ac0r871i6g0p3y47sf";
+  };
+
+  buildInputs = [ rpm cpio ];
+  propagatedBuildInputs = [ openmp ];
+
+  buildPhase = ''
+    rpm2cpio rpm/intel-mkl-common-c-${date}-${rel}-${date}-${rel}.noarch.rpm | cpio -idmv
+    rpm2cpio rpm/intel-mkl-core-rt-${date}-${rel}-${date}-${rel}.x86_64.rpm | cpio -idmv
+  '';
+
+  installPhase = ''
+    mkdir -p $out/lib
+    cp -r opt/intel/compilers_and_libraries_${version}/linux/mkl/include $out/
+    cp -r opt/intel/compilers_and_libraries_${version}/linux/mkl/lib/intel64_lin/* $out/lib/
+    cp license.txt $out/lib/
+  '';
+
+  # Per license agreement, do not modify the binary
+  dontStrip = true;
+  dontPatchELF = true;
+
+  # Some mkl calls require openmp, but Intel does not add these to SO_NEEDED and
+  # instructs users to put openmp on their LD_LIBRARY_PATH.
+  setupHook = writeText "setup-hook.sh" ''
+    addOpenmp() {
+        addToSearchPath LD_LIBRARY_PATH ${openmp}/lib
+    }
+    addEnvHooks "$targetOffset" addOpenmp
+  '';
+
+  meta = with stdenvNoCC.lib; {
+    description = "Intel Math Kernel Library";
+    longDescription = ''
+      Intel Math Kernel Library (Intel MKL) optimizes code with minimal effort
+      for future generations of Intel processors. It is compatible with your
+      choice of compilers, languages, operating systems, and linking and
+      threading models.
+    '';
+    homepage = https://software.intel.com/en-us/mkl;
+    license = licenses.issl;
```

Comment:
We should probably use `licenses.unfreeRedistributable` to build this on Hydra, or extend [isUnfree](https://github.com/NixOS/nixpkgs/blob/18.03/pkgs/stdenv/generic/check-meta.nix#L45), to support arbitrary unfree redistributable licenses.

Reply:
I agree with

> We should probably use licenses.unfreeRedistributable to build this on Hydra

but not with 

> or extend isUnfree to support arbitrary unfree redistributable licenses.

because then they would be probably be installed with `allowUnfree = false;`, which is not what users expect.

Reply:
I meant adding an attribute `redistributable = true` alongside `unfree = false`; `allowUnfree = false` will not install such packages.

Reply:
That's fine, sorry I misunderstood.

Reply:
Actually, are we sure `unfreeRedistributable` is working as intended?  I've started digging in, and it looks like Hydra doesn't actually build any of these things.  I spot-checked 5-10 packages and didn't see any of them on the [hydra nixpkgs jobset](https://hydra.nixos.org/jobset/nixpkgs/trunk#tabs-jobs).  For example, consider the package [aliza](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/science/medicine/aliza/default.nix), which is nowhere to be found on Hydra despite having an `unfreeRedistributable` license.  Looking at the source code on hydra and the meta evaluator, it looks like there's no extra condition for free = false but redistributable.

There's another license, `unfreeRedistributableFirmware`, that *is* being built and distributed -- but the reason for that is that it declares itself as `free = true`.